openwisp_monitoring/check/classes/iperf.py

```diff
@@ -0,0 +1,367 @@
+import logging
+from functools import reduce
+from json import loads
+from json.decoder import JSONDecodeError
+
+from django.core.exceptions import ValidationError
+from jsonschema import draft7_format_checker, validate
+from jsonschema.exceptions import ValidationError as SchemaError
+from swapper import load_model
+
+from openwisp_controller.connection.settings import UPDATE_STRATEGIES
+
+from .. import settings as app_settings
+from .base import BaseCheck
+
+logger = logging.getLogger(__name__)
+
+Chart = load_model('monitoring', 'Chart')
+Metric = load_model('monitoring', 'Metric')
+AlertSettings = load_model('monitoring', 'AlertSettings')
+DeviceConnection = load_model('connection', 'DeviceConnection')
+
+DEFAULT_IPERF_CHECK_CONFIG = {
+    'host': {
+        'type': 'array',
+        'items': {
+            'type': 'string',
+        },
+        'default': [],
+    },
+    # username, password max_length chosen from iperf3 docs to avoid iperf param errors
+    'username': {'type': 'string', 'default': '', 'minLength': 1, 'maxLength': 20},
+    'password': {'type': 'string', 'default': '', 'minLength': 1, 'maxLength': 20},
+    'rsa_public_key': {
+        'type': 'string',
+        'default': '',
+    },
+    'client_options': {
+        'type': 'object',
+        'properties': {
+            'port': {
+                'type': 'integer',
+                'default': 5201,
+                # max, min port chosen from iperf3 docs
+                'minimum': 1,
+                'maximum': 65535,
+            },
+            'time': {
+                'type': 'integer',
+                # Sets the interval time in seconds
+                # between periodic bandwidth, jitter, and loss reports.
+                'default': 10,
+                'minimum': 1,
+                # arbitrary chosen to avoid slowing down the queue (30min)
+                'maximum': 1800,
+            },
+            'tcp': {
+                'type': 'object',
+                'properties': {
+                    'bitrate': {
+                        'type': 'string',
+                        'default': '0',
+                    }
+                },
+            },
+            'udp': {
+                'type': 'object',
+                'properties': {
+                    'bitrate': {
+                        'type': 'string',
+                        'default': '10M',
```

Comment:
shall we default this to 30 MB? I think 10 is quite low.

Reply:
Sure